File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcRelationship/IfcRelConnects/IfcRelConnectsPorts/IfcRelConnectsPorts.py
class IfcRelConnectsPorts(IfcRelConnects):
    def __init__(self):
        self.RelatingPort: IfcPort = None
        self.RelatedPort: IfcPort = None
        self.RealizingElement: IfcElement = None

    # === WHERE CLAUSES ===
    # NoSelfReference : RelatingPort :<>: RelatedPort

File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcRelationship/IfcRelConnects/IfcRelConnectsPorts/IfcRelConnectsPorts.java
public class IfcRelConnectsPorts extends IfcRelConnects {
    public IfcPort RelatingPort;
    public IfcPort RelatedPort;
    public IfcElement RealizingElement;

    // === WHERE CLAUSES ===
    // NoSelfReference : RelatingPort :<>: RelatedPort
}
